src/server/common/controller/question-page-controller/question-page-controller.js

```diff
@@ -52,19 +53,54 @@ export class QuestionPageController extends GenericPageController {
       applicationState
     )
 
+    const pageError = req.yar.get(this.errorKey)
+
+    const errorProps = {}
+    if (pageError) {
+      errorProps.pageTitle = `Error: ${this.page.title}`
+      errorProps.errors = pageError.errors
+      errorProps.errorMessages = pageError.errorMessages
+      errorProps.viewModelOptions = {
+        validate: true,
+        question: this.page.question
+      }
+
+      return h.view(this.page.view, {
+        nextPage: req.query.redirect_uri,
+        heading: this.page.heading,
+        answer: pageError
+          ? this.page.Answer.fromState(pageError?.value)
```

Comment:
We've discussed all of this in slack, but just to make sure there's a 'to fix' tick list here: 

- I believe using `fromState` here creates the potential for bugs, since `.value` (called in the postHandler) alters the data subtly (e.g trimming or substituting in empty strings).  We're missing the test coverage to prove the previous approach & this one are equivalent, so it's probably easier to store the payload in the error state, then use `new this.page.Answer(pageError.payload, applicationState)`
- we're missing `applicationState` here, which causes the radio button options (e.g.) not to be filtered correctly when displaying an error